### PR TITLE
Fix 8482: Add Object.values and Object.entries 

### LINKED ISSUE #8482: Add definition for Object.entries and Object.values in lib.es2017.d.ts
The proposal is at stage 4 now. here is the spec: https://github.com/tc39/proposal-object-values-entries

here is the definition skeleton:

```ts
interface ObjectConstructor {
    keys(o: any): string[];
    values<T>(o: { [s: string]: T }): T[];
    values(o: any): any[];
    entries<T>(o: { [s: string]: T }): [string, T][];
    entries(o: any): [string, any][];
}
```

### PR DESCRIPTION
**Pull request by yuit**
Fix #8482

### CODE REVIEW

**Comment by mhegazy** [comment]
should we add `lib.es2017.d.ts`

**Comment by yuit** [comment]
@mhegazy we probably should but then I think we should add `target` es2017 then?

**Comment by mhegazy** [comment]
we do not have an es2016 target yet.

**Comment by mhegazy** [comment]
:+1:

File: tests/cases/conformance/es2017/useObjectValuesAndEntries3.ts [hunk 1]
```diff
@@ -0,0 +1,9 @@
+// @target: es6
```
Comment (sandersn) [inline tests/cases/conformance/es2017/useObjectValuesAndEntries3.ts]: `// @lib: ess6,es2017.object` would make this file compile without errors, right?

File: tests/cases/conformance/es2017/useObjectValuesAndEntries1.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+// @target: es5
+// @lib: es5,es2017.object
+
+var o = { a: 1, b: 2 };
```
Comment (sandersn) [inline tests/cases/conformance/es2017/useObjectValuesAndEntries1.ts]: can you have an example of a mixed-type object that uses the any-based overload as well? Something like

```ts
var o2 = { a: 'first', b: 2 };
var entries = Object.entries(o); // <-- entries: [string, any][]
```

**Comment by sandersn** [comment]
:+1: after adding a test case

File: tests/cases/conformance/es2017/useObjectValuesAndEntries3.ts (hunk 1 above)
Comment (yuit) [inline tests/cases/conformance/es2017/useObjectValuesAndEntries3.ts]: Yep!

**Pull request merged**